tables/Tables/ArrayColumn.tcc

```diff
@@ -260,92 +260,101 @@ void ArrayColumn<T>::getSlice (uInt rownr,
 template<class T>
 void
 ArrayColumn<T>::getColumnCells (const RefRows & rows,
-                                const Vector<Vector<Slice> > & arraySlices,
+                                const ColumnSlicer & columnSlicer,
                                 Array<T>& destination,
                                 Bool resize) const
 {
-    // Check to see if the data request makes sense.
-
-    IPosition columnShape = shape(rows.firstRow());
+   // Calculate the shape of the destination data.  This will be
+   // [s1, s2, ..., nR] where sI are the sum of the slice elements for
+   // that axis as contained in arraySlices [i].  nR is the number of rows
+   // in the RefRows object rows.  Resize the destination array to match.
 
-    uInt nExpectedDims = arraySlices.nelements();
+   const Vector<Slicer *> dataSlicers = columnSlicer.getDataSlicers();
+   const Vector<Slicer *> destinationSlicers = columnSlicer.getDestinationSlicers();
 
-    if (columnShape.nelements() != nExpectedDims){
+   IPosition destinationShape (columnSlicer.shape());
+   destinationShape.append (IPosition (1, rows.nrows()));
 
-        String message = String::format ("Column has wrong number of dimensions; expected %d "
-                                         " had %d",
-                                         nExpectedDims,
-                                         columnShape.nelements());
-        throw (TableArrayConformanceError (message));
-    }
-
-    // Calculate the shape of the destination data.  This will be
-    // [s1, s2, ..., nR] where sI are the sum of the slice elements for
-    // that axis as contained in arraySlices [i].  nR is the number of rows
-    // in the RefRows object rows.  Resize the destination array to match.
-
-    IPosition destinationShape (columnShape.nelements() + 1);
-    destinationShape (destinationShape.nelements() - 1) = rows.nrows();
+   static const String tag ("ArrayColumn::getColumnCells (rows, slicers, ...)");
+   checkShape (destinationShape, destination, resize, tag);
 
-    for (uInt i = 0; i < arraySlices.nelements(); i++){
+   // Fill the destination array one row at a time.
+   // If rows is not sliced then rowNumbers is simply a vector of the relevant
+   // row numbers.  When sliced, rowNumbers is a triple: (start, nRows, increment).
+   // Thus we have two different ways of walking through the selected rows.
 
-        Int n = 0;
-        if (arraySlices [i].nelements () == 0){
-            n = columnShape (i); // all elements
-        }
-        else{
+   const Vector<uInt> & rowNumbers = rows.rowVector();
+   Bool useRowSlicing = rows.isSliced();
+   int row = 0;
```

Comment:
I would prefer the built in int (and bool) to the capitalized version.  Occasionally I have a problem when I'm trying to cut out piece of code out to test in a stand-alone app, but it's usually not too big a problem to just include the header file.  

However, for compatibility I can continue to use the capitalized versions.  I think an awk or sed script ought to be able to convert everything. if we ever decide to do it, although there probably more important things to deal with.